### ISSUE TITLE
Cannot parse OSX system headers due to Clang `__builtin_convertvector` "function"

### ISSUE THREAD

**Issue by lambdageek** [visq/language-c#34]
Documentation is here: https://clang.llvm.org/docs/LanguageExtensions.html#langext-builtin-convertvector

The weird thing about `__builtin_convertvector` is that it takes a type as its second argument which results in a parse error on OSX (XCode 8.3) while parsing `<emmintrin.h>`:

```
/Applications/Xcode83.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../lib/clang/8.1.0/include/emmintrin.h:395: (column 64) [ERROR]  >>> Syntax Error !
  Syntax error !
  The symbol `__v2df' does not fit here.
```
(where `__v2df` is defined as `typedef double __v2df __attribute__ ((__vector_size__ (16)));`).

----

The `language-c` lexer is context-sensitive and makes a `CTokTyIdent` for typedef'd identifiers and a `CTokIdent` otherwise.  The parser for `argument_expression_list` (and ultimately, `primary_expression`)  does not expect `tyident` tokens.

**Comment by lambdageek** [comment]
What I've learned today is that using a context sensitive lexer is apparently one of the ways you write a C compiler, so `language-c` is just following in a long tradition of such things.

It is known as [The lexer hack](https://en.wikipedia.org/wiki/The_lexer_hack) and it is needed to disambiguate expressions like `(A)*b` which is either a cast or a multiplcation depending on whether `A` is a typename or a variable, respectively.  (I also learned that Clang and GCC now [have handwritten recursive descent parsers](http://eli.thegreenplace.net/2012/07/05/how-clang-handles-the-type-variable-name-ambiguity-of-cc) and handle the context-sensitivity differently)

So just allowing `tyident` in an argument list in general is probably non-trivial. We will probably need to parse this builtin with a special case.
